File: Prime.java
import java.util.ArrayList;
public class Prime {
  public static boolean test(int i) {
    for(int n=2;n<i;n++){
      if(i%n == 0) {
        return false;
      }
    }
    return true;
  }

  public static void main(String args[]) {
    if(args.length == 0){
      System.out.println("please call this method with a range argument ex \"java Prime 1000\"");
      System.exit(0);
    }
    int range = Integer.parseInt(args[0]);
    ArrayList<Integer> primes = new ArrayList<>();
    for(int i=0;i<range;i++){
      if(test(i)) {
        primes.add(i);
      }
    }
    System.out.printf("%d \n", primes.size());
  }
}

File: prime.rb
#!/bin/ruby
def test(prime)
  for i in 2...prime
    if(prime % i == 0) 
      return false
    end
  end
  return true
end

def main
  if ARGV[0] == nil
    puts 'please call me like "ruby prime.rb range"'
    exit 1
  else
    @range = ARGV[0].to_i
  end
  primes = []
  for i in 0...@range
    if(test(i)) 
      primes.push(i)
    end
  end
  puts primes.length
end

main
